tests/graph/test_steiner_tree.py

```diff
@@ -151,7 +151,7 @@ def test_steiner_graph_multigraph(self):
     def test_not_connected_steiner_tree(self):
         self.graph.add_node(None)
         with self.assertRaises(ValueError):
-            rustworkx.steiner_tree(self.graph, [1, 2, 8], weight_fn=float)
+            rustworkx.steiner_tree(self.graph, [1, 2, 0], weight_fn=float)
```

Comment:
It became a panic on out of bounds `Vec` access because 8 isn't actually in the graph. The `graph.add_node(None)` is taking the removed node 0's index. I guess I should update the python side to validate that all the provided terminal nodes are valid indices

Reply:
I added a check that provided terminal nodes are valid and raise a value error if they're not in: https://github.com/Qiskit/rustworkx/pull/1103/commits/17d04d14b85b30cf7c8437db8c13298a6df56959